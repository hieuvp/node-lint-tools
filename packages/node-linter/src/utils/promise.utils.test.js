const { sleep } = require('./promise.utils');

describe('sleep', () => {
  jest.useFakeTimers();

  ['str', () => {}, {}, [], false, undefined, null].forEach(args => {
    it(`should throw an error because "${args}" is not a number`, () =>
      expect(() => sleep(args)).toThrowErrorMatchingSnapshot());
  });

  [0, -10, NaN].forEach(args => {
    it(`should throw an error because "${args}" is not a positive number`, () =>
      expect(() => sleep(args)).toThrowErrorMatchingSnapshot());
  });

  [10, 100, 1000].forEach(args => {
    it(`should be able to sleep peacefully in ${args} ms`, () => {
      const promise = sleep(args);
      jest.advanceTimersByTime(args);
      return expect(promise).resolves.toBe(undefined);
    });
  });
});
