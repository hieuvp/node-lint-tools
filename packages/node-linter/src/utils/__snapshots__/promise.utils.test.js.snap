// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sleep should throw an error because "" is not a number 1`] = `"Expected \\"Number\\", instead got \\": object\\""`;

exports[`sleep should throw an error because "() => {}" is not a number 1`] = `"Expected \\"Number\\", instead got \\"() => {}: function\\""`;

exports[`sleep should throw an error because "[object Object]" is not a number 1`] = `"Expected \\"Number\\", instead got \\"[object Object]: object\\""`;

exports[`sleep should throw an error because "-10" is not a positive number 1`] = `"Millisecond must be a positive number"`;

exports[`sleep should throw an error because "0" is not a positive number 1`] = `"Millisecond must be a positive number"`;

exports[`sleep should throw an error because "NaN" is not a positive number 1`] = `"Millisecond must be a positive number"`;

exports[`sleep should throw an error because "false" is not a number 1`] = `"Expected \\"Number\\", instead got \\"false: boolean\\""`;

exports[`sleep should throw an error because "null" is not a number 1`] = `"Expected \\"Number\\", instead got \\"null: object\\""`;

exports[`sleep should throw an error because "str" is not a number 1`] = `"Expected \\"Number\\", instead got \\"str: string\\""`;

exports[`sleep should throw an error because "undefined" is not a number 1`] = `"Expected \\"Number\\", instead got \\"undefined: undefined\\""`;
